### ISSUE TITLE
[ANDROID] Crashes appeared around March 13th (Chrome 73)

### ISSUE THREAD

**Issue by dlindstrm** [react-native-webview/react-native-webview#429]
We have encountered an issue when using Android 9 in combination with Chrome 73. When wrapping a WebView within a View styled with `overflow: hidden` the entire app crashes. I solved it by setting `opacity: 0.99` on the WebView, and keep the `overflow: hidden` style on the wrapper. I found the solution [here](https://stackoverflow.com/questions/54131875/screen-blinks-once-when-rendering-a-webview-on-android/54719378#comment97153077_54719378).

This problem will most likely be more and more common since more and more people are getting Chrome 73 each day.

I hope this issue will be useful to some people that are experiencing the same problem. However, this is not an elegant solution and I would be very happy if someone with more Android knowledge could give it a shot and try to fix the real problem.

Thanks!

**Comment by Titozzz** [comment]
Can you either provide a clear repro // or tell me what's happening inside the device logs (using logcat or pidcat)?

**Comment by dlindstrm** [comment]
@Titozzz The device logs didn't give us that much:
```
Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR)
...
backtrace:
#00  pc 00000000016341cc  /data/app/com.android.chrome-EkPqPbaUdv_NQr_2fMTO4A==/base.apk
```

To reproduce it you have to:
1. Have an Android 9 device.
2. Use Chrome 73.0.3683.75. I installed Chrome Beta from Play Store and switched webview implementation in android dev settings.
3. Render this:
```
<View style={{ flex: 1, overflow: "hidden" }}>
  <WebView source={{ uri: "https://facebook.github.io/react-native/" }} />
</View>
```

**Comment by Salakar** [comment]
@lukebaker: @Titozzz and I tried to reproduce with a Samsung S10+ with Chrome version `73.0.3683.75` and couldn't reproduce this with the following snippet:

```jsx
        <View style={{ flex: 1,  overflow: 'hidden' }}>
          <WebView source={{ uri: 'https://google.com' }}/>
        </View>
```

Please could you help/advise more on how to reproduce?

Thanks

**Comment by lukebaker** [comment]
@Salakar @Titozzz I've created a sample app that crashes on the following device: Android 9; Nokia 6.1 with Chrome 73.0.3683.75.

https://github.com/lukebaker/WebViewScrollBug

Steps to reproduce should be:
1. clone repo
2. yarn install
3. react-native run-android
4. scroll past WebView content (the white area).

**Comment by Titozzz** [comment]
FYI: Does not crash on chrome 73.0.3683.75 android 8.0 Oneplus 3T

**Comment by Titozzz** [comment]
@nfischer would you have an idea about what's happening? Looks like webview crashes on android P when it gets out of screen and probably unmounted.

` Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 11172 (RenderThread)` is not very explicit

**Comment by nfischer** [comment]
Could someone file a report at https://goo.gl/9qkbdn? If you can include a compiled APK (and a link to source code), my teammate and I can repro (and can likely get some more information).

**Comment by Titozzz** [comment]
I'll do that asap.

**Comment by Titozzz** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=944020 @nfischer

**Comment by tijs** [comment]
Just to add some more context; we too experienced this issue once Chrome 73 started rolling out last thursday. We had about 150.000 crashes before we applied the opacity: 0.99 hack which fixed it in most cases. We did however run into an issue with that fix where it prevents the webview content from rendering correctly (scroll actions behave erratically, taps are registered in different location from the actual tap) with the fix applied. Seems to be limited to large webviews e.g. 5000px high embedded in a ScrollView. For those specific webview we now open them in a Modal which seems to 'fix' the crash even without needing the opacity: 0.99 fix. We are still in the middle of researching this so i don't have much more specific info than that but perhaps this extra info will help someone out who is running into the same issue.

**Comment by Titozzz** [comment]
@tijs this is so sad to read. Sadly this is out of my hands and we are waiting for more info from the Chromium team. You can read here all the comments https://bugs.chromium.org/p/chromium/issues/detail?id=944020#c11 until it went moved to internal threads we can't see. This also affects my app in production 😠.
Fingers crossed

**Comment by tijs** [comment]
@Titozzz don't be sad. I think this package has seen some great improvement since it has been extracted from react-native. Love where it's going and this sort of bug is not something you can prepare for.

Since webviews do not crash when loaded inside a race-native Modal component on Android 9 with Chrom 73 i still have some hope there is a fix beyond waiting for Chromium to do something though. I'd rather not wait for that :) Still think there is something in the interaction between the webview and the type of component it's embedded in specifically in react-native that is triggering this render crash.

**Comment by Titozzz** [comment]
@tijs Thanks, you made my day. 😃

**Comment by nfischer** [comment]
> until it went moved to internal threads we can't see

We aren't trying to hide the fix from you. We merged this into an auto-reported crash report, which we hide from the public by default to protect user security (the report has detailed crash stacks, which might potentially help a malicious actor trigger crashes or exploit user security). Fixes are landing ([commit](https://chromium-review.googlesource.com/c/chromium/src/+/1529925), [commit](https://chromium-review.googlesource.com/c/chromium/src/+/1529929)). The public issue should still be comment-able, and you can ask on that ticket to hear status updates (I've [pinged the thread](https://bugs.chromium.org/p/chromium/issues/detail?id=944020#c15) to ask what state the bug is in).

> This also affects my app in production

We recognize this bug affects real apps/users. Please understand this is not an intentional change, and bugs slip into every release (chromium is a relatively complex project, making it hard to verify every use case).

> still have some hope there is a fix beyond waiting for Chromium to do something though

My team considers this to be a bug in WebView, and we're working hard to solve it. If you want to try workarounds, well, that's a decision you could make. I'd recommend you consider waiting until we release a fix (you can see we triaged this [for M73](https://bugs.chromium.org/p/chromium/issues/detail?id=944020#c7), so that _should_ be a matter of days rather than a 6-week release cycle). You can help us by telling us your app's package name and how to repro, so we can verify our fix is correct. Again, please do so on the public crbug.

**As a final plug:** Android WebView has a [beta channel](https://www.chromium.org/developers/androidwebview/android-webview-beta), which allows your team to get WebView updates 6 weeks before users. You can use this as an opportunity to smoke test your app to make sure our team hasn't accidentally broken it (and if we have, [please file bugs](https://goo.gl/9qkbdn) against us).

---

Please take all of my comments with a grain of salt: I'm not promising what my team will push to production, I'm only trying to improve visibility and provide advice (based purely on the publicly available information).

**Comment by Titozzz** [comment]
@nfischer I did not mean anything bad, just described the current situation so that users don't panic. I well aware you guys are probably handling this right now 💪.

@salakar this made me think, would it be possible to write a test for detox with different chrome versions?

**Comment by Salakar** [comment]
> @Salakar this made me think, would it be possible to write a test for detox with different chrome versions?

I like the thought of this but I don't think we can 😞 not sure how we could control changing which webview to use in dev settings for example as detox only has apis to control the app

**Comment by AdamZaczek** [comment]
Just a piece of though - we did find a workaround but it's the hackiest thing I've seen in a while and it's extremely error-prone. You can implement special Android Chrome 72+ dedicated screens which will route to new stacks after Webviews are loaded and come back right after to have everything work without overflow: hidden.
Don't implement it in your codebases tho! I mean it, it will work but you are almost 100% better of waiting for a serious fix.

Thank you for noticing this issue so fast, we can't wait for a reliable fix!

**Comment by Titozzz** [comment]
FYI it appears to be fixed on chromium team side so we are now just wait for the release 😍

**Comment by malonguwa** [comment]
@Titozzz Many thanks for keep us updated!! Just wondering how does chromium team will release their fix? Does android user will need to click anything to update in their android device?

**Comment by Titozzz** [comment]
Well, it is tied to update to Chrome version so people that updated to the broken one should also update without you having to worry too much about it :).

**Comment by tijs** [comment]
We have found that the fix does not seem to be in the Chrome 74 release (beta) but the 75 (canary) version on the app store does seem to fix it! No idea when that usually lands in stable though.

**Comment by malonguwa** [comment]
@tijs hi, I thought Google team gonna fix it in the current buggy version 73 ? Whats the version this crash issue will be fixed? @nfischer @Titozzz Is there any more news ? :) Thanks a lot~

**Comment by Titozzz** [comment]
This should land on 73. If you read the thread it was tested on 73.0.3683.90 which hasn't been released 'yet'. No more info on my side 🙂

**Comment by malonguwa** [comment]
@Titozzz Many thanks for the reply! please keep us update if you got anything ~ :)

**Comment by malonguwa** [comment]
@nfischer hi, I just checked the whole thread in https://bugs.chromium.org/p/chromium/issues/detail?id=944020

Is there any update on the issue? I saw at the end, there is "Labels: ext-reporter-73", does this mean it is still waiting to be released? Could you mind to share some information to us when the fix for this issue will be released to current buggy version 73?

**Comment by Titozzz** [comment]
Looks like it was released today. Tested and fixed. Thank you @nfischer 🎉

**Comment by malonguwa** [comment]
@Titozzz Thanks for the test and update! I just give a try, it is also fixing my problem as well!!
But user need to upgrade their android chrome to the latest 73.03683.90 version. (the buggy version is 73.0.3683.75), just a remainder make sure you update the android chrome before doing the test~~

@Titozzz Maybe we should still pin this issue at the top just for little while?

@nfischer thanks so much to fix it!

**Comment by yassinecc** [comment]
@Titozzz @nfischer if I understand correctly, this crash should be fixed by upgrading Chrome? I have a similar SIGSEGV crash on an Android 9 device running Chrome 75, will try to upload a clean repro sometime this week

**Comment by nfischer** [comment]
@yassinecc that's almost certainly a completely unrelated bug. Upload a report to https://bugs.chromium.org/p/chromium/issues/entry?template=Webview+Bugs

**Comment by ChristianDein** [comment]
If we add "androidHardwareAccelerationDisabled" to the WebView tag, we get the following Error.

I/services.chedr: Background concurrent copying GC freed 99321(4MB) AllocSpace objects, 56(972KB) LOS objects, 38% free, 9MB/15MB, paused 1.053ms total 195.928ms
W/cr_AwContents: WebView.destroy() called while WebView is still attached to window.
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
Process: *******************+
java.lang.IllegalStateException: ScreenContainer is not attached under ReactRootView
at com.swmansion.rnscreens.ScreenContainer.findRootFragmentActivity(ScreenContainer.java:122)
at com.swmansion.rnscreens.ScreenContainer.getFragmentManager(ScreenContainer.java:138)
at com.swmansion.rnscreens.ScreenContainer.getOrCreateTransaction(ScreenContainer.java:143)
at com.swmansion.rnscreens.ScreenContainer.detachScreen(ScreenContainer.java:168)
at com.swmansion.rnscreens.ScreenContainer.onUpdate(ScreenContainer.java:219)
at com.swmansion.rnscreens.ScreenContainer.updateIfNeeded(ScreenContainer.java:202)
at com.swmansion.rnscreens.ScreenContainer.access$000(ScreenContainer.java:21)
at com.swmansion.rnscreens.ScreenContainer$1.doFrame(ScreenContainer.java:34)
at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:174)
at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:84)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:947)
at android.view.Choreographer.doCallbacks(Choreographer.java:761)
at android.view.Choreographer.doFrame(Choreographer.java:693)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
at android.os.Handler.handleCallback(Handler.java:873)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:6669)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

**Comment by nfischer** [comment]
> W/cr_AwContents: WebView.destroy() called while WebView is still attached to window.

This log indicates a bug in react-native-webview. I've filed issue #1161 for this. If you can repro, can you update that ticket with your steps?

The crash _may_ share the same root cause - I'm not entirely sure. The log is usually non-fatal (Android WebView tries its best to handle this scenario as NOOP), but it may be possible it can't handle it gracefully for some reason. The stack frames seem vaguely related to drawing the View hierarchy, so maybe these are the same root cause.

**Comment by omrip2** [comment]
> Just to add some more context; we too experienced this issue once Chrome 73 started rolling out last thursday. We had about 150.000 crashes before we applied the opacity: 0.99 hack which fixed it in most cases. We did however run into an issue with that fix where it prevents the webview content from rendering correctly (scroll actions behave erratically, taps are registered in different location from the actual tap) with the fix applied. Seems to be limited to large webviews e.g. 5000px high embedded in a ScrollView. For those specific webview we now open them in a Modal which seems to 'fix' the crash even without needing the opacity: 0.99 fix. We are still in the middle of researching this so i don't have much more specific info than that but perhaps this extra info will help someone out who is running into the same issue.

Did you find any workaround for this? I've the exact same problem and I can't find any solution

**Comment by terminox** [comment]
My team and I also encountered this issue and fixed it by setting `opacity: 0.99`.

Could someone explain how or why that work? Since it looks like magic/sorcery to me.

**Comment by SleeplessByte** [comment]
My guess is that `opacity: 0.99` forces the webview to be rendered on a different layer, and it's the regular "composition" that's normally failing @terminox .

**Comment by nfischer** [comment]
> My team and I also encountered this issue and fixed it by setting `opacity: 0.99`.

@terminox Could you please file a bug report at https://bugs.chromium.org/p/chromium/issues/entry?template=Webview+Bugs? As far as our team is aware, the original bug has been completely fixed and [the release came out over a year ago](https://github.com/react-native-community/react-native-webview/issues/429#issuecomment-475418869). The opacity workaround should no longer be necessary.

**Comment by SleeplessByte** [comment]
@nfischer this is 100% not solved in 9.4.0 of this library; people on expo are "stuck" with `"react-native-webview": "9.4.0",` for the time being, so unsure (of the little information you've given) if this would be a fix that's pushed to android phones via the webview update or if people need to update something here.

We thought it might be #1161, but crashes happen both on mount and unmount, also on 2016 class phones and later. The gist is, as soon as an app starts crashing because of a webview inside a scrollview, it doesn't get back into a fixed state. Uninstalling and re-installing the app _might_ work, for a while, until it starts happening.

Once we added `opacity: 0.99` it started working without crashes, on all phones.

**Comment by nfischer** [comment]
> @nfischer this is 100% not solved in 9.4.0 of this library; people on expo are "stuck" with "react-native-webview": "9.4.0", for the time being, so unsure (of the little information you've given) if this would be a fix that's pushed to android phones via the webview update or if people need to update something here.

The bug I'm referring to was in Android WebView (the project I work on), not in the react-native-webview library. WebView updates automatically on users' Android devices, your app doesn't need to do anything.

If you can reproduce this, please file a bug for the Android WebView team at https://bugs.chromium.org/p/chromium/issues/entry?template=Webview+Bugs

**Comment by tdous** [comment]
Can someone explain to me why this would suddenly start happening in an app where it did not happen once through several months of development? Several webviews all opened correctly for months, and then (day before delivery of the app), without any change to how those webviews are triggered or the general structure of the app and where they're located, it's Fatal signal 11 city. Reopening the app and the webview will load as the nav persisted, but going back out and then trying again just triggers the same crash.

None of the above wrapper view suggestions are helping.

**Comment by tdous** [comment]
I don't know the underlying cause of this, so does anyone know if it makes sense that it would suddenly start happening in an app where it did not happen once through a couple of months of development, and not consistently? Several webviews all opened correctly, and then (day before delivery of the app of course ;)), without any change to how those webviews are triggered, the general structure of the app, or the OS/device/emulator it suddenly became "Fatal signal 11" city.

Re-opening the app after crash, the navigation location is persisted and goes straight to the webview and appears to load it correctly. On the emulator the host machine's fans get rather excited, so clearly there's something going on under the hood at that point. I've not had time to try to analyze what yet (ship day as I said). At that point backing out of the webview and opening it crashes again.

What's changed since I first commented about this is a rebuild, with the code unchanged, and it's now not happening. Or at least consistently. Of course, no dev likes an occasional problem. ;) I wasn't clear from the previous comments in #430 and #429 if other people had this problem occasionally or constantly. In the previous build it was every time. So far in this one, not yet.

None of the wrapper view/opacity suggestions from here or #430 were making a dent when it was crashing every time.

I deleted my original comment and have only edited it slightly, as it was 4am at the time and I didn't want to come across as aggressive ;) I really appreciate the work you guys do. I also really want to try to understand the nature of the problem, to see if it makes sense that it could be fine and then suddenly not fine very abruptly. And, since it originally started, become usually fine again.

**Comment by farhankassam2** [comment]
In my case, I have a web view in two screens. I display a web view in a screen and the view displays properly. The problem for me occurs when I navigate back to the previous screen using react-navigation, which is meant to display another web view, but the Android app crashes unexpectedly.

Anyone know where I can find the android device logs to dive deeper into the problem. I also tried the opacity: 0.99 fix on the style of the WebView itself, like suggested in #430 , but that did not fix it for me. I put the opacity:0.99 on both the web views which are rendered in two separated components, but still no fix.


Any help would be greatly appreciated.
Thanks.

**Comment by FilipeAVS** [comment]
Adding `androidHardwareAccelerationDisabled` really hits the performance hard. The following worked for me:
```
        <View style={{ flex: 1 }}>
          <WebView style={{ opacity: 0.99, minHeight: 1 }} ... />
        </View>
```
Adding not only the `opacity` but also the `minHeight` does the trick for me.

**Comment by inspmoore** [comment]
> We have encountered an issue when using Android 9 in combination with Chrome 73. When wrapping a WebView within a View styled with `overflow: hidden` the entire app crashes. I solved it by setting `opacity: 0.99` on the WebView, and keep the `overflow: hidden` style on the wrapper. I found the solution [here](https://stackoverflow.com/questions/54131875/screen-blinks-once-when-rendering-a-webview-on-android/54719378#comment97153077_54719378).
> 
> This problem will most likely be more and more common since more and more people are getting Chrome 73 each day.
> 
> I hope this issue will be useful to some people that are experiencing the same problem. However, this is not an elegant solution and I would be very happy if someone with more Android knowledge could give it a shot and try to fix the real problem.
> 
> Thanks!

Thanks you for that! I have a web view wrapped in a View component, and the app is crashing on Android 9 and up when the web view gets unmounted! Absolutely incredible that this trick works 🤯

**Comment by farhankassam2** [comment]
> Thanks you for that! I have a web view wrapped in a View component, and the app is crashing on Android 9 and up when the web view gets unmounted! Absolutely incredible that this trick works 🤯

I noticed that the app crashes only the second time you mount a screen containing the WebView component. But the first time, it works fine. I am not sure if this happens for you too.

Another thing I noticed was that the app crash only occurs in the Android emulator but when run on a device, the app does not crash even when the second time the screen with the WebView component is loaded. That being said, make sure, to all of you facing this app crash issue, to ALSO test the app crash behaviour on your physical Android devices! You can simply connect your mobile to the computer via a cable and then enable USB Debugging and run the app on the device on WebStorm or whatever IDE you use.

**Comment by SleeplessByte** [comment]
I have not confirmed this works without the 0.99 opacity hack, and I see others are still searching for solutions. I doubt it's actually fixed/addressed.

**Comment by reedling78-togo** [comment]
This is still an issue for me using Android 11. No amount of opacity hacks are helping me ether. The web view renders fine but as soon as i change pages it crashes.

**Comment by adbs1** [comment]
> This is still an issue for me using Android 11. No amount of opacity hacks are helping me ether. The web view renders fine but as soon as i change pages it crashes.

Are you using react-navigation? If so have you tried this: https://github.com/facebook/react-native/issues/25494#issuecomment-724871241

**Comment by SleeplessByte** [comment]
I have not confirmed this works without the 0.99 opacity hack, and I see others are still searching for solutions. I doubt it's actually fixed/addressed.

**Comment by arthur-sousa** [comment]
I couldn't find the exact comment that helped me handle this in the past here's what I remember the fix was:

- Disable react navigation transition animations for Android in Webview screens: https://reactnavigation.org/docs/stack-navigator/#animationenabled. For some reason, having hardware acceleration enabled for Android (`androidLayerType="hardware"`) + transition animations enabled was what was causing the crashes for me.

**Comment by SleeplessByte** [comment]
I have not confirmed this works without the 0.99 opacity hack, and I see others are still searching for solutions. I doubt it's actually fixed/addressed.

**Comment by SleeplessByte** [comment]
I can confirm that it still happens and that the 0.99 hack still works.

**Comment by sergeymild** [comment]
app is crashing when i am trying to open any page, Samsung Galaxy Fold 3
Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x20 in tid 20060 (RenderThread), pid 19003

**Comment by SleeplessByte** [comment]
I can confirm that it still happens and that the 0.99 hack still works.

**Comment by isinha-onefc** [comment]
Still happening for me. I have small webviews embedded inside a scrollview for rendering iframe data, like social media embeds. The crash happens when the screen is already displaying the content at the top of the scrollview and the user tries to scroll up, or if the scrollview is already at the bottom and the user tries to scroll down. I cannot turn android hardware acceleration to false because then the videos on the app won't play. I tried the opacity hack, the overflow hack and multiple other things, but nothing seems to work. Tested on Samsung Galaxy S10+ (Android 12) and Samsung Galaxy Z Flip 3 (Android 11). Please someone help me with a fix because my app is a Samsung exclusive and all of my target audience is experiencing crashes.

**Comment by SleeplessByte** [comment]
This is not stale

**Comment by isinha-onefc** [comment]
I added `overScrollMode="never"` to my container `ScrollView` and that seems to have fixed the problem for me.

**Comment by SleeplessByte** [comment]
It's not stale.

**Comment by SleeplessByte** [comment]
This was probably not stale, but I'm also no longer interested in having to keep this open by some horrible bot that does nothing but generate more noise. Good luck everyone!

**Comment by mym0404** [comment]
Still alive problem should be reopened

**Comment by emintatli** [comment]
This problem still exists.

**Comment by andreialecu** [comment]
This is fixed by #2874 - not yet merged, but you can use it via `patch-package` or similar meanwhile.
